.github/workflows/ci.yml

```diff
@@ -0,0 +1,327 @@
+name: ci
+
+on:
+  pull_request:
+  push:
+    branches:
+      - master
+
+jobs:
+  test:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          # Core
+          - build: unit
+            os: ubuntu-latest
+          # Modules
+          - build: go-1.21
+            os: ubuntu-latest
+          - build: java-21
+            os: ubuntu-latest
+          - build: java-18
+            os: ubuntu-latest
+          - build: java-17
+            os: ubuntu-latest
+          - build: node-21
+            os: ubuntu-latest
+          - build: perl
+            os: ubuntu-latest
+          - build: php-8.3
+            os: ubuntu-latest
+          - build: python-3.12
+            os: ubuntu-latest
+          - build: ruby-3.3
+            os: ubuntu-latest
+          - build: wasm
+            os: ubuntu-latest
```

Comment:
Should we update this matrix to match our Dockerfiles? That would currently mean:

- [ ] Add Go 1.22
- [ ] Add Node 20
- [ ] Add PHP 8.2
- [ ] Add Python 3.11
- [ ] Add Ruby 3.2
- [ ] Add Perl 5.36 and 5.38
- [ ] Drop Perl 5.34 (default in `ubuntu-latest`)

I'm... not sure that makes sense right now. Maybe best to merge as is and discuss coverage after release. Thoughts, @arbourd ?

Reply:
Ya I'm happy to match those. 

Couple of issues with Perl and PHP with the default Actions runner:

- I haven't been able to get the community `setup-perl` action to play nice yet
- The community `setup-php` is also broken for anything other than 8.3 due to potential missing SAPI's.

Maybe net-add the missing ones and then follow up with Perl and PHP after?

Reply:
Works for me. I have some ideas for preventing our matrix from exploding, but for now, let the combinatorics explode.